### PR TITLE
Game stuff

### PR DESCRIPTION
**Pull request by andrewjmeier**
Some simple game logic stuff. Don't merge quite yet (unless you really need it). I want to write some tests for it.

### CODE REVIEW

**Comment by andrewjmeier** [comment]
Ok I added some tests. @qstearns @JamesEVerhagen @konnorbeard @kevinkim91 does someone want to take a look over this?

File: bundle.js [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+	card3.text = "Doctor's fees - Pay $50";
+	card3.action = function(player) {
+		player.money -= 50;
+	};
+
+	var card4 = Object.create(Card);
+	card4.text = "Get out of jail free - this card may be kept until needed, or sold";
+	card4.action = function(player) {
+		player.hasGetOutFreeCard = true;
+	};
```
Comment (konnorbeard) [inline bundle.js]: is it possible to have two of these at a time in monopoly?

Comment (andrewjmeier) [inline bundle.js]: yeah maybe switch that to a count of get out free cards?

File: bundle.js [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+			// or if (Object.prototype.hasOwnProperty.call(obj,prop)) for safety...
+				// TODO - link up correctly w/ property object
+				housesCount += player.properties[prop].numberOfHouses;
+				if (player.properties[prop].hasHotel) {
+					hotelsCount += 1;
+				}
+			}
+		}
+		var total = 40 * housesCount + 115 * hotelsCount;
+		player.money -= total;
```
Comment (konnorbeard) [inline bundle.js]: we might want to abstract money subtractions (like this one) to a separate function that checks to see if they can afford the deduction, and if not then prompts the player to mortgage/sell properties

File: bundle.js [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+		// TODO - pay owner 10x dice
+	};
+
+	var chance4 = Object.create(Card);
+	chance4.text = "Advance token to the nearest Railroad and pay owner twice the rental to which he/she is otherwise entitled. If Railroad is unowned, you may buy it from the Bank.";
+	chance4.action = function(player) {
+		if (player.position >= 35) {
+			// Reading RR
+			player.position = 5;
+			player.money += 200;
```
Comment (konnorbeard) [inline bundle.js]: I don't think directly changing player.position is the best way to handle these chance movements - is there a way we can do a player.move(position) or something?

File: bundle.js [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+	            } else if (this.players[this.currentPlayer].turnsInJail === 3) {
+	                this.players[this.currentPlayer].payBail();
+	                this.move();
+	            }
+	        } else {
+	            this.move();
+	        }
+	    },
+
+	    move: function() {
```
Comment (konnorbeard) [inline bundle.js]: might be easier to abstract this to take a dice roll argument so we can call move(diceRoll) and not have to refer to this.dice

File: src/game.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
+var Game = {
```
Comment (konnorbeard) [inline src/game.js]: seems like this is repeated code form above - is this intentional?

Comment (andrewjmeier) [inline src/game.js]: The code above is from the bundle.js file that gets compiled by web pack. I added it to the gitignore so it shouldn't show up anymore after this.

Comment (konnorbeard) [inline src/game.js]: oh maybe this is the npm stuff combining the classes?

File: bundle.js (hunk 3 above)
Comment (konnorbeard) [inline bundle.js]: Yeah now that I think about it, we're not handling the side effects for any of these position changes... we should be calling (pseudocode)

      board.spaces[position].perform_landing_action(player); 

which i think should occur in the player.move function

thoughts?

**Comment by konnorbeard** [comment]
@andrewjmeier lgtm see comments

**Pull request merged**